src/sdk/namespace_fs.js

```diff
@@ -2990,6 +3004,65 @@ class NamespaceFS {
         return false;
     }
 
+    /**
+     * @param {nb.NativeFSContext} fs_context
+     * @param {number} [size_hint]
+     * @returns {Promise<void>}
+     */
+    async _throw_if_low_space(fs_context, size_hint) {
+        if (!config.NSFS_SPC_CHECK_ENABLED) return;
+
+        const statfs = await nb_native().fs.statfs(fs_context, this.bucket_path);
```

Comment:
I think calling this before every operation is too much when we are serving hundreds+ operations per second.

Can we call it up to once every second (config value), and keep the last value in memory?

Perhaps changing nsfs_fs_space_cache to LRUCache would be the best way here.